### ISSUE TITLE
Globaleaks doesn't send emails 

### ISSUE THREAD

**Issue by Malarkey86** [globaleaks/globaleaks-whistleblowing-software#3770]
### What version of GlobaLeaks are you using?

4.13.16

### What browser(s) are you seeing the problem on?

All

### What operating system(s) are you seeing the problem on?

Windows, Linux

### Describe the issue

We are trying to implement globaleaks for our company. It seems that everything works correctly, except that it does not send emails.

when I go to the notification settings page and click on the "test" button. it works and the mail arrives instantly.

The problem is that emails are not sent in several cases. When someone files a complaint. Neither the recipient of the channel nor the user (if you specify your email).

It is also not possible to create a new user and click on the "Send activation link" button.

After investigating through github and others, I have tried updating and the only way sending emails works is by restarting the service in Ubuntu.

I do a service restart, the emails are sent and it doesn't work again until I repeat the action.

What I can do?

**Comment by evilaliv3** [comment]
Thank you @Malarkey86 we currently do not have evidences of bugs on this matter but we will try to add some debugging lines in the code to try to identify if a bug exists.

**Comment by gianlucagilardi** [comment]
For the time being and unless the root cause is identified and fixed, the best workaround is having some sort of watchdog checking the mail table

```
#!/bin/bash

FILENAME=/root/mailtest.txt


sqlite3 /var/globaleaks/globaleaks.db 'select * from mail' > /root/mailtest.txt

if [ -z "$(cat ${FILENAME})" ]
 then
  exit
else   
  systemctl restart globaleaks
fi
```

(I know, it is crappy but i just jotted down something to give you an idea). You have the script in the root crontab and at least  you can mitigate the problem.

**Comment by Malarkey86** [comment]
Here is my globaleaks.log after doing a restart

![globaleak](https://github.com/globaleaks/GlobaLeaks/assets/22235460/ffb8849c-be46-4562-a9f3-09daa5f8e109)

**Comment by evilaliv3** [comment]
Thank you @gianlucagilardi 

Your workaround is actually a very good solution but would work only in the case the emails are stuck in the pool.

It wont work instead if the failure is on the email generation.

As improvementi suggest you could improve it by checking if the creation_date of the email so do restart the application only if more than 1 hour has passed since the email generation.

**Comment by Malarkey86** [comment]
Could anyone give me a possible cause or solution? The previous one does not work at all.

**Comment by he-ftv** [comment]
Hi, we have the exact same issue. Test emails arrive immediately, but activation emails, new report notification and so on do not. @Malarkey86  did you find any solution? thanks a lot!

**Comment by he-ftv** [comment]
in the logs I see this error

![image](https://github.com/globaleaks/GlobaLeaks/assets/150698062/d7c40ccd-c721-41c5-9a8b-0785adfb4f0c)

**Comment by Malarkey86** [comment]
We have not found a solution and it must be something from globaleaks, but our log does not show any errors. We are evaluating other applications. IM sorry, I can not help you

**Comment by he-ftv** [comment]
Ok thank you. @evilaliv3 @gianlucagilardi Possiamo affrontare insieme il problema?

**Comment by he-ftv** [comment]
Resolved - it was our firewall that blocked it.

**Comment by breinhardt-dtec** [comment]
I can confirm this issue. I configured our own smtp server for notifications. Test mail arrived almos instantly. But mails for account activation, report creation and so on got stuck. After restarting the globaleaks service all mails were sent.

We're running Globaleaks on Debian 12, all packages are up-to-date.

**Comment by evilaliv3** [comment]
Are you are able to reproduce the issue?

do you have logs of globaleaks or of the mail server that you can share and contain some interesting output?

**Comment by breinhardt-dtec** [comment]
No, I didn't had the time yet to setup another instance to try to reproduce the issue. I'll try to do so within the next two weaks. But after I restarted the globaleaks service last week (as I commented on the issue) I didn't face any problems. Since now all mails arrived.

**Comment by evilaliv3** [comment]
Closing as done
